tests/testthat/test-project.R

```diff
@@ -23,6 +23,17 @@ test_that("Test against reference results", {
     pred_1 <- project(i, runif(100, 0.8, 1.9), si, n_days = 30)
     expect_equal_to_reference(pred_1, file = "rds/pred_1.rds", update = FALSE)
 
+
+    ## time-varying R
+    set.seed(1)
+    pred_2 <- project(i,
+                      R = list(1.5, 0.5, 2.1, .4, 1.4),
```

Comment:
Could we add a simpler test? Something like, having one time change with R of 1.5 and R of 0.5, and confirming that the projections around time 60 are near zero and that half the projections around time 30 are above 10?

Reply:
Will do.

Reply:
Added two tests for the new use-cases, including some simpler ones: 
```
    ## time-varying R (fixed within time windows)
    set.seed(1)
    pred_2 <- project(i,
                      R = c(1.5, 0.5, 2.1, .4, 1.4),
                      si = si,
                      n_days = 60,
                      time_change = c(10, 15, 20, 30),
                      n_sim = 100)
    expect_equal_to_reference(pred_2, file = "rds/pred_2.rds", update = FALSE)


    ## time-varying R, 2 periods, R is 2.1 then 0.5
    set.seed(1)
    
    pred_3 <- project(i,
                      R = c(2.1, 0.5),
                      si = si,
                      n_days = 60,
                      time_change = 40,
                      n_sim = 100)
    expect_equal_to_reference(pred_3, file = "rds/pred_3.rds", update = FALSE)

    ## time-varying R, 2 periods, separate distributions of R for each period
    set.seed(1)
    R_period_1 <- runif(100, min = 1.1, max = 3)
    R_period_2 <- runif(100, min = 0.6, max = .9)
    
    pred_4 <- project(i,
                      R = list(R_period_1, R_period_2),
                      si = si,
                      n_days = 60,
                      time_change = 20,
                      n_sim = 100)
    expect_equal_to_reference(pred_4, file = "rds/pred_4.rds", update = FALSE)    

```